dataclients/kubernetes/crd.go

```diff
@@ -0,0 +1,197 @@
+}
+
+type routeGroupItem struct {
+	Metadata *metadata       `json:"metadata"`
+	Spec     *routeGroupSpec `json:"spec"`
+}
+
+type routeGroupSpec struct {
+	// Hosts specifies the host headers, that will be matched for
+	// all routes created by this route group. No hosts mean
+	// catchall.
+	Hosts []string `json:"hosts,omitempty"`
+	// Backends specify the list of backends that can be
+	// referenced from routes. There should be in most cases only
+	// one default backend which is aplied to all routes, if no
+	// override was added to a route. A special case is Traffic
+	// Switching which will have more than one backend definition
+	// marked as default.
+	Backends []*skipperBackend `json:"backends"`
+	// Routes specifies the list of route based on path, method
+	// and predicates. It defaults to catchall, if there are no
+	// routes.
+	Routes []*routeSpec `json:"routes,omitempty"`
+}
+
+type routeSpec struct {
+	Path       string   `json:"path"`
+	Method     string   `json:"method,omitempty"`
+	Filters    []string `json:"filters,omitempty"`
+	Predicates []string `json:"predicates,omitempty"`
+	Backend    string   `json:"backend,omitempty"`
+}
+
+type algorithm int
+
+const (
+	none algorithm = iota
+	roundrobin
+	random
+	consistentHash
+)
+
+// required for parsing skipperBackend and adding type safety for
+// Algorithm and Type with skipperBackend type
+type skipperBackendParser struct {
+	Name string `json:"name"`
+	// Type is one of "service|shunt|loopback|lb|url"
+	Type string `json:"type"`
+	// Default backends will be applied to all routes without overrides
+	Default bool `json:"default"`
+	// Weight defines the traffic weight, if there are 2 or more
+	// default backends
+	Weight int `json:"weight"`
+	// URL is only used for Type url
+	URL string `json:"url"`
+	// ServiceName is only used for Type service
+	ServiceName string `json:"serviceName"`
+	// ServicePort is only used for Type service
+	ServicePort int `json:"servicePort"`
+	// Algorithm is only used for Type lb
+	Algorithm string `json:"algorithm"`
+	// Algorithm is only used for Type lb
+	Endpoints []string `json:"endpoints"`
+}
+
+// type safe version of skipperBackendParser
+// can be:
+// - *backend defined in definitions.go
+// - SpecialBackend string   // <shunt>, ..
+type skipperBackend struct {
+	Name        string
+	Type        eskip.BackendType
+	Default     bool
+	ServiceName string
+	ServicePort int
+	Algorithm   algorithm
+	Endpoints   []string
+}
+
+// adding Kubernetes specific backend types here. To be discussed.
+// The main reason is to differentiate between service and external, in a way
+// where we can also use the current global option to decide whether the service
+// should then be converted to LB. Or shall we expect the route group already
+// contain the pod endpoints, and ignore the global option for skipper?
+const (
+	serviceBackend = eskip.LBBackend + iota
+	externalURL
+)
+
+// Shall we support dynamic backend? Does it make sense in this
```

Comment:
At least the shunt is useful for our purposes, since we use it to reject tls requests or to only give 404 if the client is authenticated and 403 otherwise. Our catchall routes use PathSubtree("/"), is this rule equivalent:

```yaml
- path: /*
  configs:
    - predicates:
      - ...
    - filters:
      - <shunt>
    
```

Reply:
`<shunt>` is already supported